paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperWorldExtensions.java

```diff
@@ -23,6 +29,82 @@ public static void register() {
             return new ElementTag(world.getWorld().getNoTickViewDistance());
         });
 
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+
+            // <--[tag]
+            // @attribute <WorldTag.gateway_count>
+            // @returns ElementTag(Number)
+            // @description
+            // Returns the number of end gateway portals.
+            // Only works if the world is an end world.
+            // -->
+            WorldTag.tagProcessor.registerTag(ElementTag.class, "gateway_count", (attribute, object) -> {
+                DragonBattle battle = object.getWorld().getEnderDragonBattle();
+                if (battle == null) {
+                    attribute.echoError("Provided world is not an end world!");
+                    return null;
+                }
+                return new ElementTag(battle.getGatewayCount());
+            });
+
+            // <--[tag]
+            // @attribute <WorldTag.healing_crystals>
+            // @returns ListTag(EntityTag)
+            // @description
+            // Returns a ListTag of the healing end crystals located on top of the obsidian towers.
+            // Only works if the world is an end world.
+            // -->
+            WorldTag.tagProcessor.registerTag(ListTag.class, "healing_crystals", (attribute, object) -> {
+                DragonBattle battle = object.getWorld().getEnderDragonBattle();
+                if (battle == null) {
+                    attribute.echoError("Provided world is not an end world!");
+                    return null;
+                }
+                return new ListTag(battle.getHealingCrystals(), EntityTag::new);
+            });
+
+            // <--[tag]
+            // @attribute <WorldTag.respawn_crystals>
+            // @returns ListTag(EntityTag)
+            // @description
+            // Returns a ListTag of the respawn crystals located at the end exit portal.

```

Comment:
Nitpick, but I'd probably mention end crystals here (like in the `WorldTag.healing_crystals` meta)